docs/technical/installation_controller.md

```diff
@@ -1,64 +1,215 @@
+which import an export parameter of the current installation. Root installations are triggered with a reconcile
+annotation. Non-root installations are triggered by raising a reconcile event for them. The triggered successors will
+check in their own `Init` phase whether all preconditions for them are satisfied, because they might have other
+predecessors besides the current installation.
+
+When the successors have been successfully triggered, the reconcile of this installation is finished. 
+The controller sets the phase equal to `Succeeded` and the finished job ID equal to the job ID.
+
+#### Errors
+
+Suppose an error occurs during the installation flow (resp. deletion flow.)
+We distinguish between **normal errors** that could be resolved by a retry, and **fatal errors** for which a retry does 
+not make sense. 
+
+In case of a normal error, the controller will repeat the current phase. Earlier, already accomplished 
+phases are not repeated. If a phase needs to be retried several times, the intervals in between are growing. 
+
+In case of a fatal error, the phase changes to `Failed` (resp. `DeleteFailed`), so that the current flow is finished.
+The finished job ID is set equal to the job ID.
+
+#### Starting Another Reconcile Job
+
+Updating the spec or imports of a root installation does not yet start a new reconcile job. A reconcile annotation is 
+necessary to trigger the reconcile. This is only possible at root installations; a reconcile annotation at a non-root 
+installation has no effect.
+
+Suppose a reconcile annotation is added at a root installation while there is already a reconcile job running. 
+Then the current job is not interrupted. The handling of the reconcile annotation is postponed until the current job has 
+finished. Note that a changed spec or changed imports have an impact on the collection of exports in 
+[phase "Completing" phase](#phase-completing---collecting-exports).
+
+#### Deleting an Installation
+
+If the user deletes a root installation, the deletion flow will be started. However, a currently running installation 
+flow will be finished first. 
+
+The first delete does not require a trigger in form of a reconcile annotation; the deletion flow starts automatically.
+On the other hand, suppose a previous deletion flow has failed, and the installation is now in phase `DeleteFailed`. 
+Then the deletion flow can be restarted via a reconcile annotation. In both cases the controller generates a new 
+job ID and the deletion flow starts with phase `InitDelete`.
+
+#### Phase "InitDelete"
+
+Phase `InitDelete` starts with a check whether there still exist successors of the current installation, i.e. siblings 
+that import an export of the current installation. Successors must be deleted first. Therefore, the installation
+remains in phase `InitDelete` as long as successors exist. The phase and thus also the check are repeated in
+increasing time intervals.
```

Comment:
Not sure, but does this mean that if the deletion of a successors require much time, that the reconcile interval might become very large and therefore the deletion will need much time even if all successors are gone. Do we also need some triggering of predecessors here?

Reply:
Added to the todo list of the backlog item.